code/datums/emotes.dm

```diff
@@ -132,19 +133,29 @@
 		if(emote_type & EMOTE_VISIBLE)
 			var/list/viewers = get_mobs_in_view(7, user)
 			for(var/mob/current_mob in viewers)
+				for(var/obj/object in current_mob.contents)
+					if((object.flags_atom & USES_SEEING))
+						seeing_obj |= object
 				if(!(current_mob.client?.prefs.toggles_langchat & LANGCHAT_SEE_EMOTES))
 					viewers -= current_mob
 			run_langchat(user, viewers)
 		else if(emote_type & EMOTE_AUDIBLE)
 			var/list/heard = get_mobs_in_view(7, user)
 			for(var/mob/current_mob in heard)
+				for(var/obj/object in current_mob.contents)
+					if((object.flags_atom & USES_HEARING))
+						seeing_obj |= object
 				if(current_mob.ear_deaf)
 					heard -= current_mob
 					continue
 				if(!(current_mob.client?.prefs.toggles_langchat & LANGCHAT_SEE_EMOTES))
 					heard -= current_mob
 			run_langchat(user, heard)
 
+	for(var/obj/object as anything in seeing_obj)
+		if(object) //It's possible that it could be deleted in the meantime.
```

Comment:
in the meantime of what? this isn't a proc that utilizes sleep() or set background = TRUE, and the game is single threaded

Reply:
I just copypasted this from mob/living/say :sob: